Source/Core/Heap.js

```diff
@@ -65,6 +75,7 @@ Object.defineProperties(Heap.prototype, {
       return this._maximumLength;
     },
     set: function (value) {
+      removeTrailingReferences(this, value);
```

Comment:
I feel like it would be more clear if `removeTrailingReferences` were inlined in the if statement below.

Side note, I'm wondering what `value > 0` is for. Can we not set the maximum length to 0, unusual as that may be?